Downgrade gvisor-tap-vsock to v0.7.5

jandubois opened the pull request:
As part of the 1.17.0 release, we upgraded the gvisor-tap-vsock library to version v0.8.1. However, this version has caused issues with DNS resolution behind VPNs and other name resolution problems. To address these issues, we are planning to release a patch, and this downgrade is part of that solution.

Signed-off-by: Nino Kodabande <nkodabande@suse.com>
(cherry picked from commit ea1a76d9cf08493b9f78a2fe006c2fdc7ad943a7)

mook-as commented on src/go/guestagent/go.mod:
```diff
@@ -8,11 +8,11 @@ require (
-	github.com/docker/docker v27.4.0+incompatible
+	github.com/containernetworking/plugins v1.6.2
+	github.com/containers/gvisor-tap-vsock v0.7.5
+	github.com/docker/docker v27.5.0+incompatible
```
That's an _upgrade_. Probably fine, though?

jandubois replied:
I think it was a merge error. Since we are not merging any other Go module updates, we shouldn't be merging this one either.

mook-as commented on src/go/networking/go.mod:
```diff
@@ -35,18 +34,22 @@ require (
-	golang.org/x/crypto v0.30.0 // indirect
-	golang.org/x/exp v0.0.0-20240719175910-8a7402abbf56 // indirect
-	golang.org/x/mod v0.21.0 // indirect
+	golang.org/x/crypto v0.32.0 // indirect
```
This bump (0.30.0 to 0.32.0) is pulling in golang.org/x/sys 0.29.0

mook-as reviewed (changes requested):
I'm suspicious of a few of the bumps in here. Please double check that they're intended.

jandubois commented on src/go/networking/go.mod:
```diff
@@ -35,18 +34,22 @@ require (
 	golang.org/x/tools v0.26.0 // indirect
 	gopkg.in/check.v1 v1.0.0-20201130134442-10cb98267c6c // indirect
 	gopkg.in/yaml.v3 v3.0.1 // indirect
+	inet.af/tcpproxy v0.0.0-20220326234310-be3ee21c9fa0 // indirect; replaced to github.com/inetaf/tcpproxy (see the bottom of this go.mod file)
```
Because the new repo has an additional commit to change the module name in `go.mod`: https://github.com/inetaf/tcpproxy/commit/3ce58045626c8bc343a593c90354975e61b1817a

Nino-K committed:
Downgrade gvisor-tap-vsock to v0.7.5

As part of the 1.17.0 release, we upgraded the gvisor-tap-vsock library to version v0.8.1.
However, this version has caused issues with DNS resolution behind VPNs and other name resolution problems.
To address these issues, we are planning to release a patch, and this downgrade is part of that solution.

Signed-off-by: Nino Kodabande <nkodabande@suse.com>
(cherry picked from commit ea1a76d9cf08493b9f78a2fe006c2fdc7ad943a7)
